apps/src/templates/projects/projectsModule.js

```diff
@@ -15,7 +21,52 @@ export function selectedGallery(state, action) {
   }
 }
 
+export function projectLists(state, action) {
+  // A map from project type to array of projects, e.g.:
+  //   {
+  //     applab: [{...}, {...}, {...}]
+  //     gamelab: [{...}, {...}, {...}]
+  //     playlab: [{...}, {...}, {...}]
+  //     artist: [{...}, {...}, {...}]
+  //   }
+  state = state || {};
+  switch (action.type) {
+    case SET_PROJECT_LISTS:
+      return action.projectLists;
+    case ADD_OLDER_PROJECTS: {
+      // Append the incoming list of older projects to the existing list,
+      // removing duplicates.
+      const {projects, projectType} = action;
+      state = {...state};
+      state[projectType] = _.unionBy(state[projectType], projects, 'channel');
+      return state;
```

Comment:
the relevant thing to know about `unionBy` in this context is that it returns a new array as opposed to modifying the original one. OK, it seems like using Immutable vs. the `{...state, [...]: ...}` you suggested behave very similarly with the exception that Immutable makes it harder for you to screw things up when you go to add another reducer in the future.